tests/Doctrine/Tests/DBAL/Platforms/AbstractMySQLPlatformTestCase.php

```diff
@@ -907,4 +907,15 @@ public function testListTableForeignKeysSQLEvaluatesDatabase()
         self::assertContains('bar', $sql);
         self::assertNotContains('DATABASE()', $sql);
     }
+
+    public function testGetDefaultValueDeclarationSQLEscaped()
```

Comment:
Following my previous comment in `SchemaManagerFunctionnalTestCase`... it should be now easy to test the full mysql escape sequence:

Override the method:

```
protected function getEscapedLiteralSequence(): string
        return implode('+', [
            "\\0",          // An ASCII NUL (X'00') character
            "\\'", "''",    // Single quote
            '\\"', '""',    // Double quote
            '\\b',          // A backspace character
            '\\n',          // A new-line character
            '\\r',          // A carriage return character
            '\\t',          // A tab character
            '\\Z',          // ASCII 26 (Control+Z)
            '\\\\',         // A backslash (\) character
            '\\%',          // A percent (%) character
            '\\_',          // An underscore (_) character
        ];
}
```

Just an open idea, feel free to do as you like of course. At least this is what works for me in https://github.com/doctrine/dbal/pull/2825/commits/e57dcc0ea65b807366a23603bf1e2c2b1781c988.

Reply:
I'd remove all `testGetDefaultValueDeclarationSQLEscaped()` tests in favor of `testEscapedDefaultValueMustBePreserved()` which will run the same for all platforms. Escaping is part of the SQL syntax whose validity can only be checked against a real DB.

Reply:
The classes were `testGetDefaultValueDeclarationSQLEscaped()` appear are already full of tests against quoting/escaping. While I agree that it should ultimately be tested against a real database (which it is), I still see value in those kind of tests, because they are way faster to run, and thus debug, than functional tests. And since it does not seem to go against the trend of the project I'd leave them there.

Reply:
Tests removed.